R/deleteThis.R

```diff
@@ -0,0 +1,1789 @@
+
+  if (is.null(names(state)) || is.null(names(state$options)) ||
+      is.null(names(options)) || is.null(names(key))) {
+    return(NULL)
+  }
+
+  result <- list()
+  for (item in names(state)) {
+
+    if (item %in% names(key) == FALSE) {
+      result[[item]] <- state[[item]]
+      next
+    }
+
+    change <- .optionsChanged(state$options, options, key[[item]])
+    if (change == FALSE) {
+      result[[item]] <- state[[item]]
+    }
+
+  }
+
+  if (length(names(result)) > 0) {
+    return(result)
+  }
+
+  return(NULL)
+}
+
+# This recursive function removes all non-jsonifyable image objects from a
+# result list, while retaining the structure of said list.
+.imgToResults <- function(lst) {
+
+  if (! "list" %in% class(lst))
+    return(lst) # we are at an end node or have a non-list/custom object, stop
+
+  if (all(c("data", "obj") %in% names(lst)) && is.character(lst[["data"]])) {
+    # found a figure! remove its object!
+    lst <- lst[names(lst) != "obj"]
+  }
+
+  # recurse into next level
+  return(lapply(lst, .imgToResults))
+}
+
+# This recursive function takes a results object and extracts all the figure
+# objects from it, irrespective of their location within the nested structure.
+# It then returns a named list of image objects.
+.imgToState <- function(lst) {
+
+  result <- list()
+
+  if (!is.list(lst))
+    return(NULL) # we are at an end node, stop
+
+  if (all(c("data", "obj") %in% names(lst)) && is.character(lst[["data"]])) {
+    # Found a figure, add to the list!
+    name <- lst[["data"]]
+    result[[name]] <- list(
+      obj = lst[["obj"]],
+      width = lst[["width"]],
+      height = lst[["height"]]
+    )
+    return(result)
+  }
+
+  # Recurse into the next level (unname to avoid concatenating list names
+  # such as (name1.name2."data"))
+  return(unlist(lapply(unname(lst), .imgToState), recursive = FALSE))
+
+}
+
+
+as.modelTerms <- function(object, ...) UseMethod("as.modelTerms")
+as.modelTerms.list <- function(object) structure(object, class = "modelTerms")
+as.modelTerms.formula <- function(formula) structure(sapply(attr(terms(formula), "term.labels"), strsplit, ":"), class="modelTerms")
+formula.modelTerms <- function(modelTerms, env = parent.frame()) {
+  # Converts a modelTerms list into a one-side R formula
+  #
+  # Args:
+  #   modelTerms:  A list of interaction terms, each term being a list of variable names involved in the interaction
+  #   env:         An environement associated with the variables in the formula, see ?as.formula
+  #
+  # Value:
+  #   A formula. See ?formula
+  #
+  terms = sapply(modelTerms, function(x) paste0(unlist(x), collapse = ":"))
+  terms = terms[terms != ""]
+  formula.rhs = paste(terms, collapse = " + ")
+  if (formula.rhs != "") as.formula(paste(" ~ ", formula.rhs), env = env)
+}
+
+
+b64 <- function(x, ...) UseMethod("b64")   ## Translate names in x to 'base64'
+d64 <- function(x, ...) UseMethod("d64")   ## Untranslate names in x from 'base64'
```

Comment:
Could you double check if these aren't used in the classical meta analysis?

Reply:
I've moved the ones that are used back to common. But since the classical meta-analysis is the only analysis that uses these, it might be better to move these to that module.

Reply:
Theoretically I could see the use of these in jaspBase, but I highly doubt anyone will ever use them